tests/test_view.py

```diff
@@ -59,4 +63,238 @@ def test_pycon(self):
+        self.assertAlmostEqual(view_center[1], new_view_center[1], delta=1e-5)
+        self.assertAlmostEqual(view_center[2], new_view_center[2], delta=1e-5)
+
+
+@unittest.skipIf(GITHUB_ACTIONS, "GUI is not available in GitHub Actions")
+class ViewOrbitCameraTC(ViewCameraTB, unittest.TestCase):
+    camera_type = "orbit"
+
+    def setUp(self):
+        self.widget.resetCamera()
+
+    def test_reset(self):
+        dt = 0.01
+        initial_pos = self.pos()
+        initial_view_vector = self.view_vector()
+        initial_view_center = self.view_center()
+        initial_up_vector = self.up_vector()
+
+        self.move(x=1, y=1, z=1, dt=dt)
+        self.move(yaw=1, pitch=1, dt=dt, right_mouse_button=True)
+
+        self.assertNotEqual(self.pos(), initial_pos)
+        self.assertNotEqual(self.view_vector(), initial_view_vector)
+        self.assertNotEqual(self.view_center(), initial_view_center)
+        self.assertNotEqual(self.up_vector(), initial_up_vector)
+
+        self.widget.resetCamera()
+
+        self.assertEqual(self.pos(), initial_pos)
+        self.assertEqual(self.view_vector(), initial_view_vector)
+        self.assertEqual(self.view_center(), initial_view_center)
+        self.assertEqual(self.up_vector(), initial_up_vector)
+
+    def test_translation(self):
+        dt = 0.01
+        delta = dt * self.linear_speed()
+        delta_vec = np.array([delta, delta, -delta])
+        new_pos = np.array(self.pos()) + delta_vec
+        new_view_center = np.array(self.view_center()) + delta_vec
+
+        self.move(x=1, dt=dt)
+        self.assertEqual(self.pos()[0], new_pos[0])
+
+        self.move(y=1, dt=dt)
+        self.assertEqual(self.pos()[1], new_pos[1])
+
+        # camera moves in negative z direction
+        self.move(z=1, dt=dt)
+        self.assertEqual(self.pos()[2], new_pos[2])
+
+        # camera view center should move with camera
+        view_center = self.view_center()
+        self.assertEqual(view_center[0], new_view_center[0])
+        self.assertEqual(view_center[1], new_view_center[1])
+        self.assertEqual(view_center[2], new_view_center[2])
+
+    def test_rotation(self):
+        dt = 0.01
+        angle = dt * self.look_speed()
+
+        initial_view_center = self.view_center()
+        initial_view_vector = self.view_vector()
+
+        # test camera does not rotate when right mouse button is not pressed
+        self.move(yaw=1, pitch=1, dt=dt)
+        self.assertEqual(self.view_vector(), initial_view_vector)
+        self.assertEqual(self.view_center(), initial_view_center)
+
+        # test camera rotates around y-axis
+        self.move(yaw=1, dt=dt, right_mouse_button=True)
+
+        rotation_matrix = self.angle_axis(angle, (0, -1, 0))
+        rotated_vector = np.array(initial_view_vector) @ rotation_matrix
+        view_vector = self.view_vector()
+
+        self.assertAlmostEqual(rotated_vector[0], view_vector[0], delta=1e-5)
```

Comment:
[`numpy.testing.assert_allclose()`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose) can merge the 3 lines into 1, but it is not wrong to use [`assertAlmostEqual()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual).